lib/object-generator.js

```diff
@@ -80,6 +81,194 @@ var ObjectGenerator = function () {
           for (var _iterator2 = elements[Symbol.iterator](), _step2; !(_iteratorNormalCompletion2 = (_step2 = _iterator2.next()).done); _iteratorNormalCompletion2 = true) {
             var element = _step2.value;
 
+            object[element.key] = element.generateOriginalObject();
+          }
+        } catch (err) {
+          _didIteratorError2 = true;
+          _iteratorError2 = err;
+        } finally {
+          try {
+            if (!_iteratorNormalCompletion2 && _iterator2.return) {
+              _iterator2.return();
+            }
+          } finally {
+            if (_didIteratorError2) {
+              throw _iteratorError2;
+            }
+          }
+        }
+
+        return object;
+      }
+      return elements;
+    }
+
+    /**
+     * Generate a javascript object reflecting the updates from the elements.
+     *
+     * @param {Array} elements - The elements.
+     *
+     * @returns {Object} The javascript update object.
+     */
+
+  }, {
+    key: 'generateSetUpdate',
+    value: function generateSetUpdate(elements) {
+      if (elements) {
+        var object = {};
+        var _iteratorNormalCompletion3 = true;
+        var _didIteratorError3 = false;
+        var _iteratorError3 = undefined;
+
+        try {
+          for (var _iterator3 = elements[Symbol.iterator](), _step3; !(_iteratorNormalCompletion3 = (_step3 = _iterator3.next()).done); _iteratorNormalCompletion3 = true) {
+            var element = _step3.value;
+
+            if (!element.isRemoved() && element.currentKey !== '' && element.isModified()) {
+              // Include the full modified element. This way if a nested field
+              // has been altered (changed/added/removed) it is set at the top
+              // most level and exists in the set update document.
+              object[element.currentKey] = element.generateObject();
+            }
+          }
+        } catch (err) {
```

Comment:
this is quite a bit of duplication, I think we can refactor it in a function that has a good name

Reply:
Ah I think that's the `lib` file which is generated - they're checked into the repo. Here's the same method from `src`: https://github.com/mongodb-js/hadron-document/pull/71/files/adb6f357bfba7fe094d9d273751051662387680d#diff-c5b07fee6b95360111f7aa50295e7229R36